core/src/main/java/org/jdbi/v3/core/BaseStatement.java

```diff
@@ -55,24 +55,36 @@ protected StatementBuilder getStatementBuilder()
         return statementBuilder;
     }
 
-    void addCleanable(Cleanable cleanable)
+    /**
+     * Registers the given {@link Cleanable} to be executed when this statement is closed.
+     *
+     * @param cleanable the cleanable to register
+     * @return this
+     */
+    @SuppressWarnings("unchecked")
+    public final This addCleanable(Cleanable cleanable)
     {
-        getContext().getCleanables().add(cleanable);
+        getContext().addCleanable(cleanable);
+        return (This) this;
```

Comment:
Or we could hoist the `typedThis` stuff up from `SqlStatement` to `BaseStatement`